sycl/doc/extensions/ExtendedAtomics/SYCL_INTEL_extended_atomics.asciidoc

```diff
@@ -0,0 +1,525 @@
+
+Because the interfaces defined by this specification are not final and are subject to change they are not intended to be used by shipping software products.
+
+== Version
+
+Built On: {docdate} +
+Revision: 1
+
+== Contact
+John Pennycook, Intel (john 'dot' pennycook 'at' intel 'dot' com)
+
+== Dependencies
+
+This extension is written against the SYCL 1.2.1 specification, Revision v1.2.1-6.
+
+== Overview
+
+The SYCL atomic library (+cl::sycl::atomic+) defined in SYCL 1.2.1 is based on the standard atomic libary (+std::atomic+) but has some differences.  This extension introduces an alternative atomic class (+cl::sycl::intel::atomic+) including additional features from {cpp}11 and {cpp}20:
+
+- Overloaded operators to reduce the verbosity of using atomics
+- Support for floating-point types
+
+=== Overloaded Operators
+
+In SYCL 1.2.1, the +cl::sycl::atomic+ class provides atomic operations by way of member functions (e.g. +fetch_add+) without defining the corresponding operators (e.g. `+=`).  This increases the verbosity of simple uses of atomics, and requires developers to change their kernel code when switching between +read_write+ and +atomic+ accessors:
+
+[source,c++]
+----
+q.submit([&](handler& cgh)
+{
+  auto acc = buf.get_access<access::mode::atomic>(cgh);
+  cgh.parallel_for(..., [=](id<1> i)
+  {
+    acc[i].fetch_add(1); // developer cannot write acc[i]++ without this extension
+  });
+});
+----
+
+The operators defined by this extension match those defined for +std::atomic+ in {cpp}11.  The functionality of each operator is equivalent to calling a corresponding member function of the atomic class -- the operators do not expose any new functionality of the class, but act as shorthands for common use-cases.
+
+==== Operators for All Supported Types
+
+[source,c++]
+----
+  operator T();
+  T operator=(T desired);
+  T operator+=(T operand);
+  T operator-=(T operand);
+----
+
+==== Operators for Integral Types
+
+[source,c++]
+----
+  T operator++(int operand);
+  T operator--(int operand);
+  T operator++();
+  T operator--();
+  T operator&=(T operand);
+  T operator|=(T operand);
+  T operator^=(T operand);
+----
+
+=== Support for Floating-point Types
+
+In SYCL 1.2.1, support for floating-point types is limited to the +load+, +store+ and +exchange+ member functions.  Many applications requiring additional atomic operations (e.g. addition) currently work around this restriction using type punning and integer +compare_exchange+ operations.
+
+This extension extends support for floating-point types to the +compare_exchange+, +fetch_add+ and +fetch_sub+ functions in line with {cpp}20.  These new functions do not require dedicated floating-point atomic instructions and can be emulated using integer operations, giving compilers the freedom to choose the best implementation for the target device.
```

Comment:
`fetch_max` and `fetch_min` seem to be also enabled for floating-point types by this extension. Shall we mention them in this description as well?

Reply:
Yes, those were missed by mistake.  Thanks!